util/dockerutil/docker.go

```diff
@@ -91,21 +92,56 @@ func DockerPullLocalImages(ctx context.Context, fe containerutil.ContainerFronte
 
 func dockerPullLocalImage(ctx context.Context, fe containerutil.ContainerFrontend, localRegistryAddr string, pullName string, finalName string) error {
 	fullPullName := fmt.Sprintf("%s/%s", localRegistryAddr, pullName)
+
 	err := fe.ImagePull(ctx, fullPullName)
 	if err != nil {
-		return errors.Wrapf(err, "image pull")
+		return errors.Wrap(err, "image pull")
+	}
+
+	// Fix for #2471 where Podman pulls seem exit before the image is available
+	// for tagging. Wait for the image to become available.
+	err = waitForImage(ctx, fe, fullPullName)
+	if err != nil {
+		return err
 	}
+
 	err = fe.ImageTag(ctx, containerutil.ImageTag{
 		SourceRef: fullPullName,
 		TargetRef: finalName,
 	})
 	if err != nil {
 		return errors.Wrap(err, "image tag after pull")
 	}
+
 	force := true // Sometimes Docker GCs images automatically (force prevents an error).
 	err = fe.ImageRemove(ctx, force, fullPullName)
 	if err != nil {
 		return errors.Wrap(err, "image rmi after pull and retag")
 	}
+
 	return nil
 }
+
+func waitForImage(ctx context.Context, fe containerutil.ContainerFrontend, fullName string) error {
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Second)
+	defer cancel()
+
+	ticker := time.NewTicker(100 * time.Millisecond)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-ticker.C:
+			m, err := fe.ImageInfo(ctx, fullName)
+			if err != nil {
+				// Not available. Retry.
+				continue
+			}
+			if info, ok := m[fullName]; ok && info.ID != "" {
+				return nil
+			}
+		}
+	}
+}
```

Comment:
I wonder if this slows down Docker too a little bit, especially if there are multiple images being output. Could this function be refactored a bit such that it first runs `ImageInfo`, and then waits for 100ms intervals? (Instead of starting with a 100ms wait?)

Reply:
My mistake. I keep forgetting that tickers delay before the first value. Thanks for pointing that out.